concert/experiments/addons.py

```diff
@@ -385,7 +385,11 @@ async def _check_timestamp(self, producer):
         self.timestamp_incorrect = False
         i = 0
         async for item in producer:
-            timestamp = Timestamp(item)
+            if item.base is None:
+                img = item
+            else:
+                img = item.base
+            timestamp = Timestamp(img)
```

Comment:
Clever, I like it. You could just save a lot of space by `timestamp = Timestamp(item if item.base is None else item.base)`. To me that's also easily readable, but it's also personal taste I guess.